src/bp/ui-admin/src/routes/index.js

```diff
@@ -39,7 +39,19 @@ export const makeMainRoutes = () => {
           />
           <Route path="/login" render={props => <LoginPage auth={auth} {...props} />} />
           <Route path="/register" render={props => <RegisterPage auth={auth} {...props} />} />
+          <Route
+            path="/setToken"
+            render={props => {
+              const userToken = ('; ' + document.cookie)
+                .split('; userToken=')
+                .pop()
+                .split(';')
+                .shift()
```

Comment:
not sure whats going on in this block

Reply:
It allows to extract an element from the cookies, i didn't feel like adding a new library only to manage one cookie

Reply:
well yes, you might want to extract this in a function or even better in a `withToken` HOC that will compute this and provide it down to the desired component

Reply:
i've extracted it, an HOC would not be useful since it's only used in a specific case